src/tabset/tabset.ts

```diff
@@ -83,10 +94,22 @@ export class NgbTabset implements AfterContentChecked {
    */
   @Input() type: string = 'tabs';
 
+  /**
+   * A tab change event fired right before the tab selection happens
+   */
+  @Output() change = new EventEmitter<NgbTabChangeEvent>();
+
   select(tabIdx: string) {
     let selectedTab = this._getTabById(tabIdx);
-    if (selectedTab && !selectedTab.disabled) {
-      this.activeId = selectedTab.id;
+    if (selectedTab && !selectedTab.disabled && this.activeId !== selectedTab.id) {
+      let preventSelection = false;
```

Comment:
In the DOM this is named `defaultPrevented` and IMO it would be good to use the same name: https://developer.mozilla.org/en-US/docs/Web/API/Event/defaultPrevented